x/ccv/provider/keeper/proposal.go

```diff
@@ -466,73 +447,52 @@ func (k Keeper) BeginBlockCCR(ctx sdk.Context) {
 	k.DeletePendingConsumerRemovalProps(ctx, propsToExecute...)
 }
 
-// ConsumerRemovalPropsToExecute iterates over the pending consumer removal proposals
-// and returns an ordered list of consumer removal proposals to be executed,
-// ie. consumer chains to be stopped and removed from the provider chain.
+// GetConsumerRemovalPropsToExecute returns an ordered list of pending consumer
+// removal proposals that are ready to be executed,
+// i.e., consumer chains to be stopped and removed from the provider chain.
 // A prop is included in the returned list if its proposed stop time has passed.
 //
 // Note: this method is split out from BeginBlockCCR to be easily unit tested.
-func (k Keeper) ConsumerRemovalPropsToExecute(ctx sdk.Context) []types.ConsumerRemovalProposal {
-
-	// store the (to be) executed consumer removal proposals in order
+func (k Keeper) GetConsumerRemovalPropsToExecute(ctx sdk.Context) []types.ConsumerRemovalProposal {
 	propsToExecute := []types.ConsumerRemovalProposal{}
 
-	k.IteratePendingConsumerRemovalProps(ctx, func(stopTime time.Time, prop types.ConsumerRemovalProposal) (stop bool) {
-		if !ctx.BlockTime().Before(stopTime) {
+	k.IteratePendingConsumerRemovalProps(ctx, func(prop types.ConsumerRemovalProposal) (stop bool) {
+		if !ctx.BlockTime().Before(prop.StopTime) {
 			propsToExecute = append(propsToExecute, prop)
 			return false // do not stop the iteration
 		}
-		// No more proposals to check, since they're stored/ordered by timestamp.
-		return true // stop
+		// stop iteration; proposals are ordered by stop time,
+		// so no additional pending props are ready to act upon
+		return true
 	})
 
 	return propsToExecute
 }
```

Comment:
I suppose we only really need to store the time and the chainID and we only really need to return the list of chainIDs. But it's good not to change it like that in this PR 👍